File: app/src/main/java/com/example/ronasit/api/Wind.java

package com.example.ronasit.api;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Wind {

    @SerializedName("speed")
    @Expose
    private Integer speed;
    @SerializedName("deg")
    @Expose
    private Integer deg;

    /**
     * No args constructor for use in serialization
     *
     */
    public Wind() {
    }

    /**
     *
     * @param deg
     * @param speed
     */
    public Wind(Integer speed, Integer deg) {
        super();
        this.speed = speed;
        this.deg = deg;
    }

    public Integer getSpeed() {
        return speed;
    }

    public void setSpeed(Integer speed) {
        this.speed = speed;
    }

    public Wind withSpeed(Integer speed) {
        this.speed = speed;
        return this;
    }

    public Integer getDeg() {
        return deg;
    }

    public void setDeg(Integer deg) {
        this.deg = deg;
    }

    public Wind withDeg(Integer deg) {
        this.deg = deg;
        return this;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(Wind.class.getName()).append('@').append(Integer.toHexString(System.identityHashCode(this))).append('[');
        sb.append("speed");
        sb.append('=');
        sb.append(((this.speed == null) ? "<null>" : this.speed));
        sb.append(',');
        sb.append("deg");
        sb.append('=');
        sb.append(((this.deg == null) ? "<null>" : this.deg));
        sb.append(',');
        if (sb.charAt((sb.length() - 1)) == ',') {
            sb.setCharAt((sb.length() - 1), ']');
        } else {
            sb.append(']');
        }
        return sb.toString();
    }

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result * 31) + ((this.speed == null) ? 0 : this.speed.hashCode()));
        result = ((result * 31) + ((this.deg == null) ? 0 : this.deg.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof Wind) == false) {
            return false;
        }
        Wind rhs = ((Wind) other);
        return (((this.speed == rhs.speed) || ((this.speed != null) && this.speed.equals(rhs.speed))) && ((this.deg == rhs.deg) || ((this.deg != null) && this.deg.equals(rhs.deg))));
    }

}

File: app/src/main/java/com/example/sotnikov/api/weather/Wind.kt
package com.example.sotnikov.api.weather

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Wind {
    @SerializedName("speed")
    @Expose
    var speed: Double? = null

    @SerializedName("deg")
    @Expose
    var deg: Int? = null

    /**
     * No args constructor for use in serialization
     *
     */
    constructor()

    /**
     *
     * @param deg
     * @param speed
     */
    constructor(speed: Double?, deg: Int?) : super() {
        this.speed = speed
        this.deg = deg
    }

    fun withSpeed(speed: Double?): Wind {
        this.speed = speed
        return this
    }

    fun withDeg(deg: Int?): Wind {
        this.deg = deg
        return this
    }

    fun withDirection(): String {
        if (this.deg!! in 338..360 || this.deg!! in 0..22) {
            return "северный"
        } else if (this.deg!! in 23..67) {
            return "северно-восточный"
        } else if (this.deg!! in 68..112) {
            return "Восточный"
        } else if (this.deg!! in 113..157) {
            return "юго-восточный"
        } else if (this.deg!! in 158..202) {
            return "южный"
        } else if (this.deg!! in 203..247) {
            return "юго-западный"
        } else if (this.deg!! in 248..292) {
            return "западный"
        } else if (this.deg!! in 293..337) {
            return "северо-западный"
        } else {
            return ""
        }
    }


    override fun toString(): String {
        val sb = StringBuilder()
        sb.append(Wind::class.java.name).append('@')
            .append(Integer.toHexString(System.identityHashCode(this))).append('[')
        sb.append("speed")
        sb.append('=')
        sb.append(if (speed == null) "<null>" else speed)
        sb.append(',')
        sb.append("deg")
        sb.append('=')
        sb.append(if (deg == null) "<null>" else deg)
        sb.append(',')
        if (sb[sb.length - 1] == ',') {
            sb.setCharAt(sb.length - 1, ']')
        } else {
            sb.append(']')
        }
        return sb.toString()
    }

    override fun hashCode(): Int {
        var result = 1
        result = result * 31 + if (speed == null) 0 else speed.hashCode()
        result = result * 31 + if (deg == null) 0 else deg.hashCode()
        return result
    }

    override fun equals(other: Any?): Boolean {
        if (other === this) {
            return true
        }
        if (other is Wind == false) {
            return false
        }
        val rhs = other
        return (speed === rhs.speed || speed != null && speed == rhs.speed) && (deg === rhs.deg || deg != null && deg == rhs.deg)
    }
}